src/TLuaInterpreter.cpp

```diff
@@ -10754,6 +10755,55 @@ bool TLuaInterpreter::compileAndExecuteScript(const QString& code)
     }
 }
 
+// reformats given Lua code. In case of any issues, returns the original code as-is
+// issues could be invalid Lua code or the formatter code bugging out
+QString TLuaInterpreter::formatLuaCode(const QString &code)
+{
+    if (code.isEmpty()) {
+        return code;
+    }
+    lua_State* L = pIndenterState;
+    if (!L) {
+        qDebug() << "LUA CRITICAL ERROR: no suitable Lua execution unit found.";
+        return code;
+    }
+    
+    if (!validLuaCode(code)) {
+        return code;
+    }
+
+    QString escapedCode = code;
+    // escape backslashes so we can pass \n to the function
+    escapedCode.replace(QLatin1String("\\"), QLatin1String("\\\\"), Qt::CaseInsensitive);
+    // escape quotes since we'll be using quotes to pass data to the function
+    escapedCode.replace(QLatin1String("\""), QLatin1String("\\\""), Qt::CaseInsensitive);
+    // escape newlines so they don't interpreted as newlines, but instead get passed onto the function
+    escapedCode.replace(QLatin1String("\n"), QLatin1String("\\n"), Qt::CaseInsensitive);
+
+    QString thing = QString(R"(return get_formatted_code(get_ast("%1"), {indent_chunk = '  ', right_margin = 100, max_text_width = 160, keep_comments = true}))").arg(escapedCode);
+    int error = luaL_dostring(L, thing.toUtf8().constData());
+    QString n;
+    if (error != 0) {
```

Comment:
Humm, `!= 0` is redundant to my mind - but hey I think anything that isn't zero is true - I guess my `c` heritage is showing...